sample/rules.go

```diff
@@ -117,36 +117,47 @@ func (s *RulesBasedSampler) GetSampleRate(trace *types.Trace) (rate uint, keep b
 					logger.WithFields(map[string]interface{}{
 						"rule_name": rule.Name,
 					}).Logf("could not find downstream sampler for rule: %s", rule.Name)
-					return 1, true, reason + "bad_rule:" + rule.Name, ""
+					return 1, reason + "bad_rule:" + rule.Name, ""
 				}
-				rate, keep, samplerReason, key = sampler.GetSampleRate(trace)
+				rate, samplerReason, key = sampler.GetSampleRate(trace)
 				reason += rule.Name + ":" + samplerReason
 			} else {
 				rate = uint(rule.SampleRate)
-				keep = !rule.Drop && rule.SampleRate > 0 && rand.Intn(rule.SampleRate) == 0
 				reason += rule.Name
 				s.Metrics.Histogram(s.prefix+"sample_rate", float64(rate))
-			}
-
-			if keep {
-				s.Metrics.Increment(s.prefix + "num_kept")
-			} else {
-				s.Metrics.Increment(s.prefix + "num_dropped")
 				if rule.Drop {
-					// If we dropped because of an explicit drop rule, then increment that too.
-					s.Metrics.Increment(s.prefix + "num_dropped_by_drop_rule")
+					rate = 0
```

Comment:
We don't have the rule's config in the `MakeSamplingDecision` function. I changed the logic to return `rate` as 0 to signal that it's a drop decision due to the `Drop` config